src/Services/FHIR/FhirPatientService.php

```diff
@@ -32,25 +33,33 @@ public function __construct()
     }
 
     /**
-     * This implementation parses an OpenEMR patient record to a FHIR R4 Patient resource.
+     * Parses an OpenEMR patient record to a FHIR R4 Patient resource.
+     * @param $data The OpenEMR Patient record
+     * @param $encode Indicates if the return value (FHIR R4 Patient) is encoded to a string. Defaults to true.
+     * @return The FHIR R4 Patient resource
      */
     public function parseOpenEMRRecord($data = array(), $encode = true)
     {
-        // TODO convert time to utc
-        // add simple text
-        // "text": {
-        //     "status": "generated",
-        //     "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">\n      \n      <p>Patient Donald DUCK @ Acme Healthcare, Inc. MR = 654321</p>\n    \n    </div>"
-        // }
         $patientResource = new FHIRPatient();
 
-        // @todo add display text after meta
-        $nowDate = date('Y-m-d\TH:i:s');
-        $meta = array('versionId' => '1', 'lastUpdated' => $nowDate);
+        $meta = array('versionId' => '1', 'lastUpdated' => gmdate('c'));
         $patientResource->setMeta($meta);
 
         $patientResource->setActive(true);
 
+        $narrativeText = '';
+        if (isset($data['fname'])) {
+            $narrativeText = $data['fname'];
+        }
+        if (isset($data['lname'])) {
+            $narrativeText .= ' ' . $data['lname'];
+        }
+        $text = array(
+            'status' => 'generated',
+            'div' => '<div xmlns="http://www.w3.org/1999/xhtml"> <p>' . $narrativeText . '</p></div>'
+        );
+        $patientResource->setText($text);
```

Comment:
Happy to see this.

Reply:
BTW what should be the future of narration? Should we add narration in every resource or Patient Resource is special?

Reply:
I'd say all Yash..

Reply:
Here are some links to the "resource" hierarchy which may be helpful in determining which "base FHIR fields" should be included.  Each of the resources within OpenEMR's implementation are extensions of the [Domain Resource](https://www.hl7.org/fhir/domainresource.html) which in turn extends the [Base Resource](https://www.hl7.org/fhir/resource.html).